Sources/NIOCore/NIOLoopBound.swift

```diff
@@ -30,10 +30,18 @@ public struct NIOLoopBound<Value>: @unchecked Sendable {
     @usableFromInline
     /* private */ var _value: Value
 
-    /// Initialise a ``NIOLoopBound`` to `value` with the precondition that the code is running on `eventLoop`.
+    /// Initialize a ``NIOLoopBound`` to `value` with the precondition that the code is running on `eventLoop`.
     @inlinable
     public init(_ value: Value, eventLoop: EventLoop) {
         eventLoop.preconditionInEventLoop()
+        self.init(value, uncheckedEventLoop: eventLoop)
+    }
+
+    @inlinable
+    /// Initialize a ``NIOLoopBound`` to `value` with _an assertion_ that the code is running on `uncheckedEventLoop`.
+    /// Unlike a precondition check,  ``EventLoop/assertInEventLoop(file:line:)`` only performs the check in debug configuration, so the check is free in release configuration.
+    public init(_ value: Value, uncheckedEventLoop eventLoop: EventLoop) {
```

Comment:
A few hours after I pushed this up, I realized that having an unchecked `init` does not necessarily win the user that much time — you init once. Perhaps if you have to init a lot of objects that makes sense, but `uncheckedValue` seems more useful, in comparison.

Reply:
`uncheckedValue` is basically `UnsafeTransfer` which just puts the trust into the users hand that they now where things are going and that it is safe to do so.

Reply:
I think this is worth keeping: while you only `init` once, you typically shouldn't _store_ one of these: instead, you should use it to effect a transfer.